### PR TITLE
Create unit tests for magnification API

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
None

### Summary of the issue:
There are no unit tests for the magnification API

### Description of user facing changes
None

### Description of development approach
Certain magnification functions that were not included in Windows versions older than Windows 8 had special handling.
This handling can be removed now that the minimum support version is Windows 8.

Creates basic tests for unit testing the Magnification API, starting with screen curtain functions

### Testing strategy:
run unit tests
### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

@coderabbitai summary

### CODE REVIEW

File: tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+from visionEnhancementProviders.screenCurtain import Magnification, TRANSFORM_BLACK
+
+
+class Test_ScreenCurtain(unittest.TestCase):
+	def setUp(self):
+		self.assertTrue(Magnification.MagInitialize())
+
+	def tearDown(self):
+		self.assertTrue(Magnification.MagUninitialize())
```
Comment (SaschaCowley) [inline tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py:18]: Can the setup and teardown methods be put in a common base class?

**Review by SaschaCowley — APPROVED** [review]

File: tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		for i in range(5):
+			for j in range(5):
+				self.assertEqual(TRANSFORM_BLACK.transform[i][j], resultEffect.transform[i][j])
+
+	def test_getDefaultIdentityFullscreenColorEffect(self):
+		resultEffect = Magnification.MagGetFullscreenColorEffect()
+		for i in range(5):
+			for j in range(5):
+				# The transform matrix should be the identity matrix
+				self.assertEqual(int(i == j), resultEffect.transform[i][j], f"i={i}, j={j}")
```
Comment (SaschaCowley) [inline tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py:33]: Is it worth putting this in a subtest?

File: tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	def tearDown(self):
+		self.assertTrue(Magnification.MagUninitialize())
+
+	def test_setAndConfirmBlackFullscreenColorEffect(self):
+		result = Magnification.MagSetFullscreenColorEffect(TRANSFORM_BLACK)
+		self.assertTrue(result)
+		resultEffect = Magnification.MagGetFullscreenColorEffect()
+		for i in range(5):
+			for j in range(5):
+				self.assertEqual(TRANSFORM_BLACK.transform[i][j], resultEffect.transform[i][j])
```
Comment (SaschaCowley) [inline tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py:26]: Is it worth putting this in a subtest?

File: tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py (hunk 1 above)
Comment (seanbudd) [reply tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py:18]: I'd prefer not to as I expect the setup and tear down to potentially diverge as more tests are added

Comment (seanbudd) [reply tests/unit/test_visionEnhancementProviders/test_magnificationAPI.py:18]: actually, that will only really affect the magnifier test classes, not these

**Review by SaschaCowley — APPROVED** [review]
Thanks, @seanbudd

**Comment by CyrilleB79** [comment]
@seanbudd, running these new tests with Windows Magnifier active or screen curtain active fails. Is it expected? It's a quite bad UX.

I acknowledge that for system tests, some constraints such as running them without Magnifier or screen curtain could be accepted, but for unit test, it's really strange.

What do you think.

**Pull request merged**